syntax="proto3";
import "tracking.proto";

package xn_adp.ml;
option java_package = "com.xiaoniuhy.adp.pb.ml";
option java_multiple_files = true;

message CandidateAd
{
    int32 planid = 1;     // 计划ID
    int32 campid = 2; // 广告组ID
    int32 advid = 3;  // 广告主ID
    int32 landingtype = 4;    // 投放类型：APP，LINK
    int32 actiontype = 6;     // 交互类型：APK下载，打开页面
    string category = 7;     // 广告分类
};
message PredictRequest
{
    xn_adp.tracking.BidRequest bidreq = 1;
    repeated CandidateAd ads = 2;
};

message PredictResult
{
    int32 planid = 1;
    int32 pctr = 2;     // 预估ctr * 10^6
    int32 pcvr = 3;     // 预估cvr * 10^6
};
message PredictResponse
{
    int32 code = 1;
    string msg = 2;
    repeated  PredictResult data = 3;
};